pkg/tsdb/cloudwatch/response_parser.go

```diff
@@ -66,13 +67,42 @@ func parseGetMetricDataTimeSeries(metricDataResults map[string]*cloudwatch.Metri
 	result := tsdb.TimeSeriesSlice{}
 	partialData := false
 	metricDataResultLabels := make([]string, 0)
-	for k := range metricDataResults {
-		metricDataResultLabels = append(metricDataResultLabels, k)
+
+	if requestAzs, ok := query.Dimensions["AvailabilityZone"]; ok {
```

Comment:
This problem can certainly happen for all kinds of dimensions, so just checking AvailabilityZone will not suffice.